DSharpPlus.Commands/Processors/TextCommands/TextCommandProcessor.cs

```diff
@@ -64,15 +67,64 @@ public override async ValueTask ConfigureAsync(CommandsExtension extension)
         await base.ConfigureAsync(extension);
     }
 
-    public async Task ExecuteTextCommandAsync(DiscordClient client, MessageCreatedEventArgs eventArgs)
+    internal async Task ExecuteTextCommandAsync(DiscordClient client, MessageUpdatedEventArgs eventArgs)
+    {
+        if (this.extension is null)
+        {
+            throw new InvalidOperationException("TextCommandProcessor has not been configured.");
+        }
+        else if (!this.Configuration.EnableEditCommands)
+        {
+            return;
+        }
+        else if (this.messageUpdatedCache.TryGetValue(eventArgs.Message.Id, out DiscordMessage? cachedMessage) && cachedMessage is not null)
+        {
+            // Check to see if the edited at time is the same as the cached message's edited at time.
+            // This is important because when Discord scans messages for NSFW content, it will send a
+            // message updated event.
+            if (eventArgs.Message.EditedTimestamp == cachedMessage.EditedTimestamp)
+            {
+                return;
+            }
+        }
+
+        this.messageUpdatedCache.Set(eventArgs.Message.Id, eventArgs.Message, new MemoryCacheEntryOptions()
+        {
+            SlidingExpiration = TimeSpan.FromSeconds(15)
+        });
+
+        // Check to see if the flags are the same, excluding the flags that are not relevant to the message content.
+        DiscordMessageFlags notTheseFlags = DiscordMessageFlags.Crossposted | DiscordMessageFlags.IsCrosspost | DiscordMessageFlags.SuppressedEmbeds;
+
+        // Flags are null when 0. 0 is not equal to null.
+        DiscordMessageFlags relevantFlags = eventArgs.Message.Flags & ~notTheseFlags ?? 0;
+        DiscordMessageFlags relevantFlagsBefore = eventArgs.MessageBefore?.Flags & ~notTheseFlags ?? 0;
+        if (eventArgs.MessageBefore is not null && relevantFlags == relevantFlagsBefore)

```

Comment:
Checking flags is fine but I would also check if the content is actually updated. If this is irrelevant please note I'm on mobile and will do a more thorough look through soon

Reply:
No this is right actually. Good catch